samples/videoDecode/videodecode.cpp

```diff
@@ -329,27 +326,20 @@ int main(int argc, char **argv) {
                 std::cout << std::setfill('0') << std::setw(2) << std::hex << static_cast<int>(digest[i]);
             }
             std::cout << std::endl;
-
             if (b_md5_check) {
-                char ref_md5_string[33], c2[2];
+                std::string ref_md5_string(32, 0);
                 uint8_t ref_md5[16];
-                std::string str;
-
+                ref_md5_file.open(md5_file_path.c_str(), std::ios::in);
+                ref_md5_file.getline(ref_md5_string.data(), 33);
```

Comment:
I suggest using ref_md5_string.length() instead of hardcoded value.
Also check if the return is success before using

Reply:
Done.